x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_tab_content.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { find } from 'lodash';
+import React, { FunctionComponent, useState } from 'react';
+
+import { EuiEmptyPrompt, EuiLink, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+
+import { SectionLoading } from '../../../shared_imports';
+import { GroupByOption, LevelFilterOption, UpgradeAssistantTabProps } from '../types';
+import { CheckupControls } from './controls';
+import { GroupedDeprecations } from './deprecations/grouped';
+import { EsDeprecationErrors } from './es_deprecation_errors';
+
+const i18nTexts = {
+  isLoading: i18n.translate('xpack.upgradeAssistant.esDeprecations.loadingText', {
+    defaultMessage: 'Loading deprecations…',
+  }),
+};
+
+export interface CheckupTabProps extends UpgradeAssistantTabProps {
+  checkupLabel: string;
+}
+
+/**
+ * Displays a list of deprecations that are filterable and groupable. Can be used for cluster,
+ * nodes, or indices deprecations.
+ */
+export const DeprecationTabContent: FunctionComponent<CheckupTabProps> = ({
+  checkupLabel,
+  deprecations,
+  error,
+  isLoading,
+  refreshCheckupData,
+  navigateToOverviewPage,
+}) => {
+  const [currentFilter, setCurrentFilter] = useState<LevelFilterOption>(LevelFilterOption.all);
+  const [search, setSearch] = useState<string>('');
+  const [currentGroupBy, setCurrentGroupBy] = useState<GroupByOption>(GroupByOption.message);
+
+  const changeFilter = (filter: LevelFilterOption) => {
+    setCurrentFilter(filter);
+  };
+
+  const changeSearch = (newSearch: string) => {
+    setSearch(newSearch);
+  };
+
+  const changeGroupBy = (groupBy: GroupByOption) => {
```

Comment:
Same here. Is there a specific reason for those?

Reply:
👍 Updated (this was pre-existing behavior)